subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/SubscriptionsManager.kt

```diff
@@ -93,25 +108,82 @@ class RealSubscriptionsManager @Inject constructor(
     private val authDataStore: AuthDataStore,
     private val billingClientWrapper: BillingClientWrapper,
     private val context: Context,
+    @AppCoroutineScope private val coroutineScope: CoroutineScope,
+    private val dispatcherProvider: DispatcherProvider,
 ) : SubscriptionsManager {
 
     private val adapter = Moshi.Builder().build().adapter(ResponseError::class.java)
 
+    private val _currentPurchaseState = MutableSharedFlow<CurrentPurchase>()
+    override val currentPurchaseState = _currentPurchaseState.asSharedFlow()
+
     private val _isSignedIn = MutableStateFlow(isUserAuthenticated())
     override val isSignedIn = _isSignedIn.asStateFlow()
 
+    private val _hasSubscription = MutableStateFlow(false)
+    override val hasSubscription = _hasSubscription.asStateFlow()
```

Comment:
Haven't tested this but maybe we can remove the `init` blocks below using `.onSubscription {  }`
Thos init blocks are always problematic because the add to the app launch time, as they're executed upon app creation

I was gonna test it but I got block in one of the steps (see asana task) so I don't know if it'd change behavior or not.

Reply:
Done, tested and working. No idea about the dialog not showing though, let me know if it happens again and we can debug.